examples/mqtt/README.md

```diff
@@ -0,0 +1,80 @@
+
+
+## Using MqttDeviceComms
+
+For both ProjectICE & non-ProjectICE use-cases, MQTT related parameters required in `publish()` and `subscribe()` like topic, qos, etc., are fetched directly from the user's
+configuration file.  Please refer this ![example](/examples/mqtt/iot_cc/iotcc_simulated_mqtt.py) for detailed explanation.
+
+
+## Using MqttDccComms
+
+MqttDccComms provides the flexibility of having a single topic (auto-generated or from configuration file) per MQTT connection or individual topic per metric. **MqttMessagingAttributes class** in ![mqtt.py](/liota/lib/transports/mqtt.py)
+provides this flexibility.
+
+### MqttMessagingAttributes
+
+It enables the following options for developers in LIOTA.
+
+**(a)** Use single publish and subscribe topic generated by LIOTA for an EdgeSystem, its Devices and its Metrics. By default:
+* Publish topic for all Metrics will be **liota/generated_local_uuid_of_edge_system**
+* Subscribe topic will be **liota-resp/generated_local_uuid_of_edge_system**
+* Publish and Subscribe QoS will be **One**
+* Publish retain flag will be **False** and
+* Subscribe call_back will be **None**
+
+**(b)** Use custom single publish and subscribe topic for an EdgeSystem, its Devices and Metrics.  Along with custom topics, other parameters (QoS, retain_flag ,etc.,) can be referred from ![Property File](/examples/mqtt/aws_iot/awsSampleProp.conf).
+
+**-** In the above two cases, MQTT message's payload MUST be self-descriptive so that subscriber can subscribe process accordingly to a single topic by parsing payload.
+
+**(c)** Use custom publish and subscribe topics for Metrics.
+
+**-** In this case, MQTT message's payload need not be self-descriptive.  Subscribers can subscribe to appropriate topics accordingly.
+
+**(d)** Use combination of **(a) and (c)** or **(b) and (c)**.
+
+
+**NOTE:**
+If **Option (a)** is used, `generated_local_uuid_of_edge_system` will be written to a file available at path **uuid_path** as specified in ![liota.conf](/config/liota.conf) file.
+Users can refer this file to get the uuid for subscribing.
+
+
+Configuring MqttDccComms and MessagingAttributes for ProjectICE and non-ProjectICE related use-cases is explained in the following sections.
+
+### Configuration for ProjectICE Scenario
+![IoTCC DCC](/liota/dccs/iotcc.py) will support both WebSockets and MQTT as transports.  **This is WIP**.  Once developed it'll allow plug-and-play approach to easily switch between these two protocols.  However, for MQTT
+it follows `Option (a)` as mentioned in the above section.
+* **Option (a)** could be achieved by passing **mqtt_msg_attr=None while initializing MqttDccComms**.
+
+
+### Configuration for Non-ProjectICE Scenario
+![GenericMqtt DCC](/liota/dccs/generic_mqtt.py) is a Data Center Component to connect with MQTT brokers using LIOTA in non-ProjectICE scenarios.  **Options (a), (b), (c) and (d)** mentioned in the above section
```

Comment:
Conventional MQTT is MQTT over TCP and there is also support for MQTT over WebSocket.  Most of the MQTT brokers (AWS, EMQTT, etc.,) support it. There is no difference in configurations except for parameter `transport` in MQTT configuration.

Yes.  That would be a DCC specific to AWSIoT only.  Here what we are trying to showcase is that, MQTT support in LIOTA is generic so that it can be used to publish data to any DCC that supports MQTT.  As of now, this example showcases publishing data to AWSIoT over MQTT.  Examples using Azure is ready https://github.com/Venkat2811/liota/tree/azure-example/examples/mqtt/azure_iot.  We're working on IBM Bluemix too.  Once we have support for multiple transports like, HTTP-REST, AMQP, we can provide platform specific Dcc.